openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -710,6 +706,47 @@ Cache-Control: no-store
 }
 ```
 
+# Nonce Endpoint {#nonce-endpoint}
+
+This endpoint allows a Client to acquire a fresh `c_nonce` value without the overhead of a full Credential Request. A Credential Issuer that requires `c_nonce` values to be incorporated into proofs in the Credential Request (see (#credential-request)) MUST offer a Nonce Endpoint.
+
+The `nonce_endpoint` Credential Issuer Metadata parameter, as defined in (#credential-issuer-parameters), contains the URL of the Credential Issuer's Nonce Endpoint.
+
+
+## Nonce Request {#nonce-request}
+
+A request for a nonce is made by sending an HTTP GET request to the URL provided in the `nonce_endpoint` Credential Issuer Metadata parameter.
+
+Below is a non-normative example of a Nonce Request:
+
+```
+GET /nonce HTTP/1.1
+Host: credential-issuer.example.com
+```
+
+## Nonce Response {#nonce-response}
+
+The Credential Issuer provides a nonce value in the HTTP response with a 2xx status code and the following parameters included as top-level members in the message body of the HTTP response using the application/json media type:
+
+* `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)).
+* `c_nonce_expires_in`: OPTIONAL. Number denoting the lifetime in seconds of the `c_nonce`. This value serves only as a hint to the client, indicating how long the Credential Issuer is likely to accept the `c_nonce` as valid.
+
+Due to the temporal and contextually sensitive nature of the `c_nonce` value, the Credential Issuer MUST make the response uncacheable by adding a `Cache-Control` header field including the value `no-store`.
```

Comment:
Perhaps this text could be a bit more lenient, namely to allow cases where the same `c_nonce` can be cached for a small amount of time (via the `max-age` value on the `Cache-Control`).
```suggestion
Due to the temporal and contextually sensitive nature of the `c_nonce` value, the Credential Issuer MUST explicitly define the allowed caching for the response, for instance by adding the `Cache-Control` header with the `no-store` value, if the response must be uncacheable.
```

Reply:
I don't think that should be accounted for at the HTTP layer of cache.

Reply:
I mean, sure the client will hold onto the thing so it can use it on a subsequent credential request but that's not something that should be conveyed at the HTTP cache level.

Reply:
if we allow max-age I think we just define the `nonce_expires_in` on an HTTP level

Reply:
or not at any level https://github.com/openid/OpenID4VCI/issues/394

Reply:
by which i mean, it's not appropriate at the HTTP layer and it's not needed at the application layer so why have it at all?